shims/spark301/src/main/scala/com/nvidia/spark/rapids/shims/spark301/GpuShuffleExchangeExec.scala

```diff
@@ -20,13 +20,14 @@ import org.apache.spark.sql.catalyst.plans.logical.Statistics
 import org.apache.spark.sql.catalyst.plans.physical.Partitioning
 import org.apache.spark.sql.execution.{ShufflePartitionSpec, SparkPlan}
 import org.apache.spark.sql.execution.exchange.ShuffleExchangeLike
-import org.apache.spark.sql.rapids.execution.GpuShuffleExchangeExecBase
+import org.apache.spark.sql.rapids.execution.GpuShuffleExchangeExecBaseWithMetrics
```

Comment:
This might be a stupid question: If the build is failing on Databricks 8.2 (i.e. the `3.1.1` shim), why must we change the `3.0.1` shim?

Reply:
I changed the class hierarchy to allow databricks 8.2 to diverge from the others. I did this by inserting a new class in between `GpuShuffleExchangeExecBase` and all of the other implementations. This let Databrciks modify how metrics are accessed and still have a common implementation for the others.  A similar thing also happened with broadcast for the future there too.

I thought it was better than copying hundreds of lines of code just to change a handful of them.